voxgraph/src/frontend/voxgraph_mapper.cpp

```diff
@@ -242,9 +250,11 @@ void VoxgraphMapper::loopClosureCallback(
   Transformation T_t1_t2(translation.cast<voxblox::FloatingPoint>(),
                          rotation.cast<voxblox::FloatingPoint>());
   Transformation T_AB = T_A_t1 * T_t1_t2 * T_B_t2.inverse();
-  pose_graph_interface_.addLoopClosureMeasurement(submap_id_A, submap_id_B,
-                                                  T_AB);
-
+  // Check if loop closure constraints is enabled
```

Comment:
With the current implementation of this callback, the loop closure constraints would be shown in Rviz despite not being used in the optimization.
This could be confusing to other users. It'd be cleaner if the first block of code at the top of this method was:
```
if (!loop_closure_constraints_enabled_) {
    ROS_WARN_THROTTLE(1.0, "Received loop closure constraint, but loop_closure_constraints_enabled is set to false. Constraint will be ignored.");
    return;
}
```
So that if loop closures are disabled, they're not added to the pose graph and not shown in Rviz to keep the two consistent.

Alternatively, you could start the method with:
```
if (!loop_closure_constraints_enabled_) {
    ROS_WARN_THROTTLE(1.0, "Received loop closure constraint, but loop_closure_constraints_enabled is set to false. Constraint will be ignored by the optimization but still shown in Rviz.");
}
```

But there'd still be a high chance that people don't see the warning message (e.g. when running it on a robot, or when there are many other messages being published to the same log), so unless there's a strong reason not to, I'd rather go with the first option.